src/shared/components/mutationMapper/MutationMapperUserSelectionStore.ts

```diff
@@ -0,0 +1,11 @@
+import {observable} from "mobx";
+import {TrackNames, TrackVisibility} from "../tracks/TrackSelector";
+
+export default class MutationMapperUserSelectionStore
```

Comment:
We can add more states (like column visibility) into this store to shared the state of other types of use selection. For now only supporting track visibility.

Reply:
Could we just keep these properties on view itself?

Reply:
Do you mean have an explicit `trackVisibility` property for `MutationMapper` component, and then instantiate an observable `trackVisibility` in the parent components `Mutations` and `MutationMapperTool`? In that case we need to do the same instantiation for `trackVisibility` (and for some possible future shared user selection observables) in multiple places, that's why I wanted to wrap these shared states inside a separate class.

@alisman We can still instantiate this shared store in the parent components (`Mutations` and `MutationMapperTool`) and explicitly pass the properties (currently only `trackVisibilitly`) instead of the store instance if you think it is a better way to do it.

Reply:
The basic idea is that I don't want to have lots of different stores floating nested at various layers of view heirarchy.   I can stomach multiple stores but at top level.   If there is state which is truly local to features/views then i think they can be stored on view.

Reply:
@alisman removed the store from the `MutationMapper` component props. Keeping it only at top level (in `Mutations` and `MutationMapperTool`) and explicitly passing a `TrackVisibility` prop to `MutationMapper`.